collector/hddtemp_linux.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2015 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !nohddtemp
+
+package collector
+
+import (
+	"io/ioutil"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type hddtempCollector struct {
+	temp *prometheus.GaugeVec
+}
+
+func init() {
+	registerCollector("hddtemp", true, NewHddtempCollector)
```

Comment:
Please use the `defaultDisabled`/`defaultEnabled` constants to make clear what the boolean stands for.

The hddtemp daemon is not installed by default on any of the distribution I checked. I'm not sure it makes sense to enable this collector by default.